gnucash/import-export/ofx/gnc-ofx-import.c

```diff
@@ -1038,6 +1025,95 @@ double ofx_get_investment_amount(const struct OfxTransactionData* data)
     }
 }
 
+// gnc_ofx_process_next_file processes the next file in the info->file_iter list.
+static void
+gnc_ofx_process_next_file (GtkDialog *dialog, gpointer user_data)
+{
+    ofx_info* info = (ofx_info*) user_data;
+    if (info->file_iter->next)
+    {
+        // Done with the previous OFX file, process the next one if any.
+        GSList* temp = info->file_iter;
+        info->file_iter = info->file_iter->next;
+        temp = g_slist_delete_link(temp, temp);
+        gnc_file_ofx_import_process_file (info);
+    }
+    else
+    {
+        // No more OFX file to process, final cleanup.
+        g_free (info);
+        if (ofx_created_commodites)
+        {
+            /* FIXME: Present some result window about the newly created
+             * commodities */
+            g_warning ("Created %d new commodities during import", g_list_length (ofx_created_commodites));
+            g_list_free (ofx_created_commodites);
+            ofx_created_commodites = NULL;
+        }
+    }
+}
+
+// When the user is done matching transactions, trigger processing the next OFX file.
+static void
+gnc_ofx_match_done (GtkDialog *dialog, gint response_id, gpointer user_data)
+{
+    ofx_info* info = (ofx_info*) user_data;
+    gtk_widget_hide(GTK_WIDGET(dialog));
+    gnc_ofx_process_next_file (dialog, info);
+}
+
+// Aux function to process the OFX file in info->file_iter
+static void
+gnc_file_ofx_import_process_file (ofx_info* info)
+{
+    LibofxContextPtr libofx_context = libofx_get_new_context();
+    char* filename = info->file_iter->data;
+    GtkWindow *parent = info->parent;
+
+#ifdef G_OS_WIN32
+    char * selected_filename = g_win32_locale_filename_from_utf8 (filename);
+#else
+    char * selected_filename = g_strdup (filename);
+#endif
```

Comment:
Nitpick: the g_strdup on non-Windows platforms is not really needed. The selected_filename is used as is (no changes made to it). So you can avoid the memory copy penalty here if you also make the g_free at the end Windows only.

Reply:
ah, OK I didn't realize that.

Reply:
Do you plan to change this still ?

Reply:
This has been done.